otherlibs/unix/envir.c

```diff
@@ -13,18 +13,48 @@
 /*                                                                        */
 /**************************************************************************/
 
+#include <caml/config.h>
+
+#ifdef HAS_UNISTD
+#include <unistd.h>
+#endif
+#include <sys/types.h>
+
 #include <caml/mlvalues.h>
 #include <caml/alloc.h>
 
-#ifndef _WIN32
 extern char ** environ;
-#endif
 
-CAMLprim value unix_environment(value unit)
+CAMLprim value unix_environment_unsafe(value unit)
 {
   if (environ != NULL) {
     return caml_copy_string_array((const char**)environ);
   } else {
     return Atom(0);
   }
 }
+
+static char **secure_environ(void)
+{
+#if defined(HAS_ISSETUGID)
```

Comment:
Thanks; I'd missed that discussion.  I've added a check for `getauxval` (ed55295).

My understanding is that `getauxval` subsumes the `getuid`/`geteuid` checks.  Consequently, the new implementation falls back to checking `issetugid`, `geteuid`, etc., only if `getauxval` is unavailable (rather than falling back in the case where the call to `getauxval` returns `0`).